### PR TITLE
Add Cars Endpoint.

### LINKED ISSUE #9: (3pts) - Add Api Endpoints Routes
## Add Api Endpoints Routes

We will implement the following:

:pushpin: We will define the routes structure por the api backend:

- [x] We create a route for users.

- [x] We create a route for cars.

- [x] We create a route for reservations.

### LINKED ISSUE #16: (3ptx) - Add Api Cars Endpoint
## Add Api Cars Endpoint

We will implement the following:

- [x] We will implement the api/v1/cars endpoint so the user can do GET and POST requests.

- [x] We will implement the api/v1/cars/:id endpoint so the users can do GET, PUT, and DELETE requests.

### PR DESCRIPTION
**Pull request by paulsaenzsucre**
## Add Cars Endpoint.

These are the proposed changes for this pull request:

:pushpin: We implement the api endpoints routes.

:pushpin: We implement RESTFUL /api/v1/cars endpoint, the user can do GET and POST requests.

:pushpin: We implement RESTFUL /api/v1/cars/:id endpoint, the user can do GET, PUT, and DELETE requests.

:pushpin: We proposed a standard response from api:

```
{
  "errors": false,
  "message_code": "founded",
  "message": "Car founded",
  "payload": {
    "id": 3,
    "model": "Mustang",
    "image": "https://www.someurl.jpeg",
    "price": "32500.0",
    "brand": "Ford",
    "year": "2021-01-01",
    "color": "Red",
    "accidents": 1,
    "rented": true,
    "electric": false
  }
}
```
If the request were successful, you have `"errors" = false` otherwise it would be `"errors" = true` and inside payload you have the resource requested to the API if any.

Requests templates:

image

image

image

image

image

### CODE REVIEW

File: app/controllers/api/v1/cars_controller.rb [hunk 1]
```diff
@@ -21,6 +21,27 @@ def create
     end
   end
 
+  # PUT /cars/:id
+  def update
+    if @car
+      @car.update(car_params)
```
Comment (Juank628) [inline app/controllers/api/v1/cars_controller.rb]: Please, can you confirm what happens if the update method fails? According to the code, the client will receive a success message and only receives an error message if the car doesn't exist.

File: app/controllers/api/v1/cars_controller.rb [hunk 2]
```diff
@@ -21,6 +21,27 @@ def create
...
+    else
+      render json: { error: 'Unable to update car.' }, status: :unprocessable_entity
+    end
+  end
+
+  # DELETE /cars/:id
+  def destroy
+    @car = Car.find(params[:id])
+    if @car
+      @car.destroy
```
Comment (Juank628) [inline app/controllers/api/v1/cars_controller.rb]: Please, can you confirm what happens if the destroy method fails? According to the code, the client will receive a success message and only receives an error message if the car doesn't exist.

**Review by Juank628 — CHANGES_REQUESTED** [review]
Great job!!! but, please, try to add some lines of code to send an error message if the update and destroy methods fails.

File: app/controllers/api/v1/cars_controller.rb (hunk 1 above)
Comment (paulsaenzsucre) [reply app/controllers/api/v1/cars_controller.rb]: Yes!!! You're right, I'm improving it

File: app/controllers/api/v1/cars_controller.rb (hunk 2 above)
Comment (paulsaenzsucre) [reply app/controllers/api/v1/cars_controller.rb]: Yes!!! You're right, I'm improving it

File: app/controllers/api/v1/cars_controller.rb (hunk 1 above)
Comment (paulsaenzsucre) [reply app/controllers/api/v1/cars_controller.rb]: I fixed. Now when you tried to update a car that doesn't exits the api answer with a message:

image

And when you tried to update an attribute that doesn't exists the api answer with another message:

image

And when you tried to update an atribute with an invalid value the api answer with another message:

image

File: app/controllers/api/v1/cars_controller.rb (hunk 2 above)
Comment (paulsaenzsucre) [reply app/controllers/api/v1/cars_controller.rb]: I fixed. Now when you tried to delete a car that doesn't exits the api answer with a message:

image

**Review by Juank628 — APPROVED** [review]
Great job!!! Now you can merge!!!

**Pull request merged**